backend/internal/oauth/oauth2/introspect/handler.go

```diff
@@ -64,15 +58,11 @@ func (h *tokenIntrospectionHandler) HandleIntrospect(w http.ResponseWriter, r *h
 	if token == "" {
 		w.Header().Set(serverconst.ContentTypeHeaderName, serverconst.ContentTypeJSON)
 		w.WriteHeader(http.StatusBadRequest)
-
 		errResp := model.ErrorResponse{
 			Error:            constants.ErrorInvalidRequest,
 			ErrorDescription: "Token parameter is required",
 		}
-		if err := json.NewEncoder(w).Encode(errResp); err != nil {
-			logger.Error("Error encoding error response", log.Error(err))
-			http.Error(w, "Failed to encode error response", http.StatusInternalServerError)
-		}
+		_ = json.NewEncoder(w).Encode(errResp)
```

Comment:
Let's keep the original implementation here. Not good to neglect  the encoding error. And this will not solved by the fix we are planning to do after this refactoring as well.